tests/gateway/test_integration.py

```diff
@@ -0,0 +1,370 @@
+                    "15 million degrees Celsius (27 million degrees Fahrenheit).",
+                },
+                "metadata": {"finish_reason": "stop"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": 17,
+            "output_tokens": 24,
+            "total_tokens": 41,
+            "model": "gpt-3.5-turbo-0301",
+            "route_type": "llm/v1/chat",
+        },
+    }
+
+    data = {"messages": [{"role": "user", "content": "How hot is the core of the sun?"}]}
+
+    async def mock_chat(self, payload):
+        await asyncio.sleep(0.1)
+        return expected_output
+
+    monkeypatch.setattr(OpenAIProvider, "chat", mock_chat)
+
+    response = query(route=route.name, data=data)
+    assert response == expected_output
+
+
+def test_openai_completions(gateway, monkeypatch):
+    client = MlflowGatewayClient(gateway_uri=gateway.url)
+    route = client.get_route("completions-openai")
+    expected_output = {
+        "candidates": [
+            {
+                "text": "OK, this is a mocked test.",
+                "metadata": {"finish_reason": "stop"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": 4,
+            "output_tokens": 7,
+            "total_tokens": 11,
+            "model": "gpt-4",
+            "route_type": "llm/v1/completions",
+        },
+    }
+
+    data = {"prompt": "mock my test", "max_tokens": 50}
+
+    async def mock_completions(self, payload):
+        await asyncio.sleep(0.1)
+        return expected_output
+
+    monkeypatch.setattr(OpenAIProvider, "completions", mock_completions)
+
+    response = client.query(route=route.name, data=data)
+    assert response == expected_output
+
+
+def test_openai_embeddings(gateway, monkeypatch):
+    client = MlflowGatewayClient(gateway_uri=gateway.url)
+    route = client.get_route("embeddings-openai")
+    expected_output = {
+        "embeddings": [[0.1, 0.2, 0.3]],
+        "metadata": {
+            "input_tokens": 4,
+            "output_tokens": 0,
+            "total_tokens": 4,
+            "model": "text-embedding-ada-002",
+            "route_type": "llm/v1/embeddings",
+        },
+    }
+
+    data = {"text": "mock me and my test"}
+
+    async def mock_embeddings(self, payload):
+        await asyncio.sleep(0.1)
+        return expected_output
+
+    monkeypatch.setattr(OpenAIProvider, "embeddings", mock_embeddings)
+
+    response = client.query(route=route.name, data=data)
+    assert response == expected_output
+
+
+def test_anthropic_completions(gateway, monkeypatch):
+    set_gateway_uri(gateway_uri=gateway.url)
+    route = get_route("completions-anthropic")
+    expected_output = {
+        "candidates": [
+            {
+                "text": "Here are the steps for making a peanut butter sandwich:\n\n1. Get bread. "
+                "\n\n2. Spread peanut butter on bread.\n\n3. Eat sandwich.",
```

Comment:
We could use it if this test needs this exact text for testing purposes, otherwise we should use a random string like `"test"` so that we don't need to wonder "why this text?".